.github/workflows/ci.yml

```diff
@@ -127,7 +127,8 @@ jobs:
     steps:
       - uses: actions/checkout@v3
       - uses: dtolnay/rust-toolchain@nightly
-      - run: ./ci/run_ci.sh rust
+      - name: Run Rust CI
```

Comment:
Could we add windows ci to validate the script?

Reply:
Add macos-latest and windows-2022 ?

Reply:
CI can run successfully on macos, but I encountered some problems on windows. I don't know much about rust, and the errors generated are as follows.

This error should have nothing to do with the script.

```
2024-01-11 07:01:09,510 - INFO - command result: 
2024-01-11 07:01:09,510 - INFO - running command: cargo test
   Compiling fury-derive v0.0.1 (D:\a\incubator-fury\incubator-fury\rust\fury-derive)
   Compiling fury v0.0.1 (D:\a\incubator-fury\incubator-fury\rust\fury)
warning: unused import: `ArrayViewer`
  --> fury\src\row\mod.rs:23:28
   |
23 | pub use reader::{from_row, ArrayViewer, StructViewer};
   |                            ^^^^^^^^^^^
   |
   = note: `#[warn(unused_imports)]` on by default

warning: unused import: `ArrayWriter`
  --> fury\src\row\mod.rs:25:26
   |
25 | pub use writer::{to_row, ArrayWriter, StructWriter};
   |                          ^^^^^^^^^^^

warning: function `compute_string_hash` is never used
   --> fury\src\types.rs:187:8
    |
187 | pub fn compute_string_hash(s: &str) -> u32 {
    |        ^^^^^^^^^^^^^^^^^^^
    |
    = note: `#[warn(dead_code)]` on by default

warning: constant `IS_NULL_FLAG` is never used
   --> fury\src\types.rs:237:15
    |
237 |     pub const IS_NULL_FLAG: u8 = 1 << 0;
    |               ^^^^^^^^^^^^

warning: constant `IS_OUT_OF_BAND_FLAG` is never used
   --> fury\src\types.rs:240:15
    |
240 |     pub const IS_OUT_OF_BAND_FLAG: u8 = 8;
    |               ^^^^^^^^^^^^^^^^^^^

warning: `fury` (lib) generated 5 warnings (run `cargo fix --lib -p fury` to apply 2 suggestions)
   Compiling fury-tests v0.0.1 (D:\a\incubator-fury\incubator-fury\rust\tests)
warning: `fury` (lib test) generated 5 warnings (5 duplicates)
    Finished test [unoptimized + debuginfo] target(s) in 1.95s
     Running unittests src\lib.rs (target\debug\deps\fury-40a37b81d76b42a1.exe)
     Running tests\de.rs (target\debug\deps\de-ada5dc3a110[335](https://github.com/LiangliangSui/incubator-fury/actions/runs/7485191914/job/20373598441#step:4:336)74.exe)
     Running unittests src\lib.rs (target\debug\deps\fury_derive-221793791d0278b7.exe)
     Running tests\test_complex_struct.rs (target\debug\deps\test_complex_struct-beb71868ebd28abd.exe)
     Running tests\test_row.rs (target\debug\deps\test_row-3b837135e9a5fc8c.exe)
error: test failed, to rerun pass `-p fury-tests --test test_row`
2024-01-11 07:01:11,608 - ERROR - 
running 0 tests

test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s


running 0 tests

test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s


running 0 tests

test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s


running 3 tests
test encode_to_obin ... ok
test decode_py_struct ... ok
test complex_struct ... ok

test result: ok. 3 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s


running 1 test
test row ... FAILED

failures:

---- row stdout ----
thread 'row' panicked at fury\src\row\reader.rs:105:28:
slice index starts at 17592186044460 but ends at 80
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace


failures:
    row

test result: FAILED. 0 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s
```

Reply:
@theweipeng Could you take a look?

Reply:
> @theweipeng Could you take a look?
These warn are OK and will be dealt with later.

Reply:
```
failures:

---- row stdout ----
thread 'row' panicked at fury\src\row\reader.rs:105:28:
slice index starts at 17592186044460 but ends at 80
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```
Hi @theweipeng , is there any solution for the above test failure on Windows?

Reply:
@LiangliangSui  Thanks for catching this!  That's my problem. I didn't notice failures before. But I am not accessable to a windows device this moment, I will submit another PR with a fix later.